### ISSUE TITLE
don't ignore type annotations on top-level vals that turn into a "let"

### ISSUE THREAD

**Issue by JonathanAldrich** [wyvernlang/wyvern#86]
Write a test case that fails because annotations on a top-level "val" are ignored.  An example is Bool.wyv without the "launderAsInt" call.  Then add the (optional) ability for the "let" construct in the core IL to have an annotation on the let-bound variable, and fill in this annotation when the source code specifies it.  Thus get the test case to pass.

**Comment by craigthelinguist** [comment]
Am working on this one. Should have a PR this week.

**Comment by JonathanAldrich** [comment]
Oops...this actually got fixed somewhere along the way.  I changed Bool.wyv just now to illustrate that the issue is solved.  I'm sorry for not marking this bug as solved--I think it got solved as part of some other change.

**Comment by craigthelinguist** [comment]
I think there might still be a few kinks. If you delete the end of Bool.wyv so the last line is this one:

`val testFive : Int = cond.ifTrue(() => 5,() => 2)`

The test fails because Bool.wyv will now type to Dyn.

Here's another example. The following types to Int and outputs 5:

```
val something : Dyn = 5
val five : Int = something
five
```
But this program types to Dyn:

```
val something : Dyn = 5
val five : Int = something
```

Is that how it should be?

**Comment by JonathanAldrich** [comment]
Yes, this is an issue with the translation of declarations, when a declaration is the last thing in a file.
